File: src/main/resources/statics/js/modules/unimall/unimallordersku.js
$(function () {
    $("#jqGrid").jqGrid({
        url: baseURL + 'unimall/unimallordersku/list',
        datatype: "json",
        colModel: [			
			{ label: 'id', name: 'id', index: 'id', width: 50, key: true },
			{ label: '', name: 'skuId', index: 'sku_id', width: 80 }, 			
			{ label: '', name: 'spuId', index: 'spu_id', width: 80 }, 			
			{ label: '', name: 'orderId', index: 'order_id', width: 80 }, 			
			{ label: '', name: 'orderNo', index: 'order_no', width: 80 }, 			
			{ label: '', name: 'spuTitle', index: 'spu_title', width: 80 }, 			
			{ label: '', name: 'title', index: 'title', width: 80 }, 			
			{ label: '', name: 'barCode', index: 'bar_code', width: 80 }, 			
			{ label: '', name: 'num', index: 'num', width: 80 }, 			
			{ label: '', name: 'unit', index: 'unit', width: 80 }, 			
			{ label: '', name: 'originalPrice', index: 'original_price', width: 80 }, 			
			{ label: '', name: 'price', index: 'price', width: 80 }, 			
			{ label: '', name: 'img', index: 'img', width: 80 }, 			
			{ label: '', name: 'gmtUpdate', index: 'gmt_update', width: 80 }, 			
			{ label: '', name: 'gmtCreate', index: 'gmt_create', width: 80 }			
        ],
		viewrecords: true,
        height: 385,
        rowNum: 10,
		rowList : [10,30,50],
        rownumbers: true, 
        rownumWidth: 25, 
        autowidth:true,
        multiselect: true,
        pager: "#jqGridPager",
        jsonReader : {
            root: "page.list",
            page: "page.currPage",
            total: "page.totalPage",
            records: "page.totalCount"
        },
        prmNames : {
            page:"page", 
            rows:"limit", 
            order: "order"
        },
        gridComplete:function(){
        	//隐藏grid底部滚动条
        	$("#jqGrid").closest(".ui-jqgrid-bdiv").css({ "overflow-x" : "hidden" }); 
        }
    });
});

var vm = new Vue({
	el:'#icloudapp',
	data:{
		showList: true,
		title: null,
		unimallOrderSku: {}
	},
	methods: {
		query: function () {
			vm.reload();
		},
		add: function(){
			vm.showList = false;
			vm.title = "新增";
			vm.unimallOrderSku = {};
		},
		update: function (event) {
			var id = getSelectedRow();
			if(id == null){
				return ;
			}
			vm.showList = false;
            vm.title = "修改";
            
            vm.getInfo(id)
		},
		saveOrUpdate: function (event) {
		    $('#btnSaveOrUpdate').button('loading').delay(1000).queue(function() {
                var url = vm.unimallOrderSku.id == null ? "unimall/unimallordersku/save" : "unimall/unimallordersku/update";
                $.ajax({
                    type: "POST",
                    url: baseURL + url,
                    contentType: "application/json",
                    data: JSON.stringify(vm.unimallOrderSku),
                    success: function(r){
                        if(r.code === 0){
                             layer.msg("操作成功", {icon: 1});
                             vm.reload();
                             $('#btnSaveOrUpdate').button('reset');
                             $('#btnSaveOrUpdate').dequeue();
                        }else{
                            layer.alert(r.msg);
                            $('#btnSaveOrUpdate').button('reset');
                            $('#btnSaveOrUpdate').dequeue();
                        }
                    }
                });
			});
		},
		del: function (event) {
			var ids = getSelectedRows();
			if(ids == null){
				return ;
			}
			var lock = false;
            layer.confirm('确定要删除选中的记录？', {
                btn: ['确定','取消'] //按钮
            }, function(){
               if(!lock) {
                    lock = true;
		            $.ajax({
                        type: "POST",
                        url: baseURL + "unimall/unimallordersku/delete",
                        contentType: "application/json",
                        data: JSON.stringify(ids),
                        success: function(r){
                            if(r.code == 0){
                                layer.msg("操作成功", {icon: 1});
                                $("#jqGrid").trigger("reloadGrid");
                            }else{
                                layer.alert(r.msg);
                            }
                        }
				    });
			    }
             }, function(){
             });
		},
		getInfo: function(id){
			$.get(baseURL + "unimall/unimallordersku/info/"+id, function(r){
                vm.unimallOrderSku = r.unimallOrderSku;
            });
		},
		reload: function (event) {
			vm.showList = true;
			var page = $("#jqGrid").jqGrid('getGridParam','page');
			$("#jqGrid").jqGrid('setGridParam',{ 
                page:page
            }).trigger("reloadGrid");
		}
	}
});

File: src/main/java/com/icloud/modules/unimall/entity/UnimallOrderSku.java
package com.icloud.modules.unimall.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import lombok.Data;

import java.io.Serializable;
import java.util.Date;

/**
 * 
 * 
 * @author zdh
 * @email yyyyyy@cm.com
 * @date 2020-05-15 15:33:01
 */
@Data
@TableName("t_unimall_order_sku")
public class UnimallOrderSku implements Serializable {
	private static final long serialVersionUID = 1L;

   	   /*  */
       @TableId(value="id", type= IdType.AUTO)
       private Long id;
   	   	   /*  */
       @TableField("sku_id")
       private Long skuId;
   	   	   /*  */
       @TableField("spu_id")
       private Long spuId;
   	   	   /*  */
       @TableField("order_id")
       private Long orderId;
   	   	   /*  */
       @TableField("order_no")
       private String orderNo;
   	   	   /*  */
       @TableField("spu_title")
       private String spuTitle;
   	   	   /*  */
       @TableField("title")
       private String title;
   	   	   /*  */
       @TableField("bar_code")
       private String barCode;
   	   	   /*  */
       @TableField("num")
       private Integer num;
   	   	   /*  */
       @TableField("unit")
       private String unit;
   	   	   /*  */
       @TableField("original_price")
       private Integer originalPrice;
   	   	   /*  */
       @TableField("price")
       private Integer price;
   	   	   /*  */
       @TableField("img")
       private String img;
   	   	   /*  */
       @TableField("gmt_update")
       private Date gmtUpdate;
   	   	   /*  */
       @TableField("gmt_create")
       private Date gmtCreate;
   	
}
